api/model/src/main/java/org/projectnessie/model/ContentKey.java

```diff
@@ -165,22 +166,35 @@ public static ContentKey of(@JsonProperty("elements") List<String> elements) {
   @Value.Check
   protected void validate() {
     List<String> elements = getElements();
+    int sum = 0;
     for (String e : elements) {
       if (e == null) {
         throw new IllegalArgumentException(
-            String.format("Content key '%s' must not contain a null element.", elements));
+            format("Content key '%s' must not contain a null element.", elements));
       }
-      if (e.isEmpty()) {
+      int l = e.length();
+      sum += l;
```

Comment:
Since we are micro-optimizing, this statement could be moved to line 189.

Reply:
That would optimize for the bad case though.